src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx

```diff
@@ -0,0 +1,203 @@
+    break;
+  case 'DONE':
+    // sign up is complete, user can be taken to an authenticated state.
+    break;
+}
+```
+
+```dart
+// v1
+final result = await Amplify.Auth.signUp(/* ... */);
+switch (result.nextStep.signUpStep) {
+  case AuthSignUpStep.confirmSignUp:
+    // prompt user to confirm sign in using nextStep.codeDeliveryDetails
+    break;
+  case AuthSignUpStep.done:
+    // sign up is complete, user can be taken to an authenticated state.
+    break;
+}
+```
+
+- Auth hub events are now have a specific sub type, `AuthHubEvent` and contain a `type` that is an enum. See the example below for how to migrate.
+
+```dart
+// v0
+final hubSubscription = Amplify.Hub.listen([HubChannel.Auth], (hubEvent) {
+  switch(hubEvent.eventName) {
+    case 'SIGNED_IN':
+      safePrint('User is signed in.');
+      break;
+    case 'SIGNED_OUT':
+      safePrint('User is signed out.');
+      break;
+    case 'SESSION_EXPIRED':
+      safePrint('The session has expired.');
+      break;
+    case 'USER_DELETED':
+      safePrint('The user has been deleted.');
+      break;
+  }
+});
+```
+
+```dart
+// v1
+final subscription = Amplify.Hub.listen(HubChannel.Auth, (AuthHubEvent event) {
+  switch (event.type) {
+    case AuthHubEventType.signedIn:
+      safePrint('User is signed in.');
+      break;
+    case AuthHubEventType.signedOut:
+      safePrint('User is signed out.');
+      break;
+    case AuthHubEventType.sessionExpired:
+      safePrint('The session has expired.');
+      break;
+    case AuthHubEventType.userDeleted:
+      safePrint('The user has been deleted.');
+      break;
+  }
+});
+```
+
+- Cognito specific API options such as `CognitoSignInOptions` are deprecated in favor of `pluginOptions`. The classes will be removed in a future release. See below for an example of how to migrate for `Auth.signIn()`.
+
+```dart
+// v0
+final result = await Amplify.Auth.signIn(
+  username: username,
+  password: password,
+  options: const CognitoSignInOptions(
+    clientMetadata: {'data': 'value'},
+  ),
+);
+```
+
+```dart
+// v1
+final result = await Amplify.Auth.signIn(
+  username: username,
+  password: password,
+  options: const SignInOptions(
+    pluginOptions: CognitoSignInPluginOptions(
+      clientMetadata: {'data': 'value'},
+    ),
+  ),
+);
+```
+
+## Analytics
+
+- `AnalyticsProperties` renamed to `CustomProperties`
+- `AnalyticsUserProfile` renamed to `UserProfile`
+- `AnalyticsUserProfileLocation` renamed to `UserProfileLocation`
+- In v0, the `autoFlushEventsInterval` in `amplifyconfiguration.dart` was read as seconds on iOS and milliseconds on Android. v1 fixes this behavior by reading `autoFlushEventsInterval` as seconds across all platforms.
+- in v1 flushing occurs at 30 second intervals. In v0 default auto flush iOS was 60 seconds, Android was 30 seconds.
+- in v1 session pause and resume are not automatically tracked. In v0 iOS auto tracked session pause and resume events.
```

Comment:
```suggestion
- In v1, session pause and resume events are not automatically tracked. In v0, iOS had auto-tracked session pause and resume events.
```

Reply:
What does this mean, though? As a customer, I would wonder how to enable it or if I need it.

Reply:
@fjnoyp ?

Reply:
One sec, looking at this

Reply:
- In v1, to ensure consistent platform behavior, backgrounding and foregrounding the app will track and trigger a new session.  This is different than v0's behavior on iOS, which tracks session Pause and Resume events on app background/foreground.

Reply:
Let's use Jordan's suggested rephrasing for this:

There was a platform inconsistency in v0. iOS paused the session when the app went into background, whereas Android stopped it and then started a new session on app resume. In v1, the behavior will match android from v0.

Reply:
v0 behavior: iOS tracked session resume/pause. iOS and Android tracked session start/stop events. 
v1 behavior: All platforms track session resume/pause.

Reply:
Apologies for the delay here.  Thinking this rephrase could be good: 

Automatic session reporting behavior has been aligned in v1.  For all platforms, backgrounding and foregrounding the app will stop and start a new session.  Previously on iOS, the session would pause and then resume.